exp/clusterresourceset/controllers/clusterresourceset_controller.go

```diff
@@ -0,0 +1,380 @@
+		switch resource.Kind {
+		case string(clusterresourcesetv1.ConfigMapClusterResourceSetResourceKind):
+			// Set status in ClusterResourceSetBinding in case of early continue due to a failure. Do this for only supported kinds.
+			clusterResourceSetInMap.Resources[GenerateResourceKindName(resource)] = clusterresourcesetv1.ClusterResourceSetResourceStatus{
+				Hash:            "",
+				Successful:      false,
+				LastAppliedTime: &metav1.Time{Time: time.Now().UTC()},
+			}
+
+			resourceConfigMap, err := GetConfigMapFromNamespacedName(context.Background(), r.Client, typedName)
+			if err != nil {
+				logger.Error(err, "Failed to find ConfigMap resource", "ConfigMap name", typedName)
+				errList = append(errList, err)
+				continue
+			}
+
+			for _, dataStr := range resourceConfigMap.Data {
+				data := []byte(dataStr)
+				dataList = append(dataList, data)
+			}
+		case string(clusterresourcesetv1.SecretClusterResourceSetResourceKind):
+			// Set status in ClusterResourceSetBinding in case of early continue due to a failure. Do this for only supported kinds.
+			clusterResourceSetInMap.Resources[GenerateResourceKindName(resource)] = clusterresourcesetv1.ClusterResourceSetResourceStatus{
+				Hash:            "",
+				Successful:      false,
+				LastAppliedTime: &metav1.Time{Time: time.Now().UTC()},
+			}
+
+			resourceSecret, err := secret.GetAnySecretFromNamespacedName(context.Background(), r.Client, typedName)
+			if err != nil {
+				logger.Error(err, "Failed to find Secret resource", "Secret name", typedName)
+				errList = append(errList, err)
+				continue
+			}
+
+			if resourceSecret.Type != clusterresourcesetv1.ClusterResourceSetSecretType {
+				logger.Error(err, "Unsupported ClusterResourceSet Secret type", "Resource type", resourceSecret.Type)
+				errList = append(errList, err)
+				continue
+			}
+			for _, dataStr := range resourceSecret.Data {
+				dataList = append(dataList, dataStr)
+			}
+
+		default:
+			logger.Error(err, "Unsupported ClusterResourceSet resource kind", "Resource Kind", resource.Kind)
+			errList = append(errList, err)
+			continue
+		}
+
+		// Apply all values in the key-value pair of the resource to the cluster.
+		// As there can be multiple key-value pairs in a resource, each value may have multiple objects in it.
+		isSuccessful := true
+		for i := range dataList {
+			data := dataList[i]
+
+			if err := Apply(ctx, c, data); err != nil {
+				isSuccessful = false
+				logger.Error(err, "failed to apply ClusterResourceSet resource %q/%q in namespace %q", "Resource kind", resource.Kind, "Resource name", resource.Name)
+				errList = append(errList, err)
+			}
+		}
+
+		resourceInMap = *GenerateResourceStatus(isSuccessful, ComputeHash(dataList))
+		clusterResourceSetInMap.Resources[GenerateResourceKindName(resource)] = resourceInMap
+	}
+	if len(errList) > 0 {
+		return kerrors.NewAggregate(errList)
```

Comment:
Is it possible to set this aggregated error in ClusterResourceSetBinding or ClusterResourceSet status. Currently, nothing is set in status or binding if an error happens and the only way to debug is using logs

Reply:
In the CAEP, there were suggestions about using Conditions. I am looking into that if it will be enough.